core/src/main/java/hudson/util/io/Archiver.java

```diff
@@ -36,11 +39,17 @@
  */
 public abstract class Archiver extends FileVisitor implements Closeable {
     protected int entriesWritten =0;
+    protected List<String> files = new ArrayList<>();
+
 
     /**
      * Number of files/directories archived.
      */
     public int countEntries() {
         return entriesWritten;
     }
+
+    public List<String> getFiles() {
```

Comment:
nit: Would be good to add some docs:

```suggestion
    /**
     * A list of the relative paths of all of the files in the archive.
     */
    public List<String> getFiles() {
```